### ISSUE TITLE
No cast function found for return type object_array

### ISSUE THREAD

**Issue by devoncrouse** [crate/crate#1631]
Java 1.7.0_60
Crate 0.45.8

```bash
# Create source table with object array, and insert one record
crash -c "create table source_urls (urls array(object(dynamic) as (uri string, count integer))) with (number_of_replicas = 0);"
crash -c "insert into source_urls (urls) values ([{ uri='blah.com', count=1 }]);"

# Attempt to insert value into an identical destination table
crash -c "create table dest_urls (urls array(object(dynamic) as (uri string, count integer))) with (number_of_replicas = 0);"
crash -c "insert into dest_urls (urls)(select urls from source_urls)"
```

```
SQLActionException[No cast function found for return type object_array]
```

**Comment by globalspin** [comment]
I looked for a workaround, but there doesn't seem to be a combination of cast functions that end up with the data type the insert needs.
```
cr> select cast(urls as string) from source_urls;
SQLActionException[type 'object_array' not supported for conversion]
cr> select cast(urls as array(string)) from source_urls;
SQLActionException[Array inner type 'object' not supported for conversion]
```
Oddly, it is capable of rendering the object array as a string, but that might just be crash doing the conversion:

```
cr> select urls from source_urls;
+--------------------------------------+
| urls                                 |
+--------------------------------------+
| [{"count": 1, "uri": "example.com"}] |
+--------------------------------------+
```

**Comment by chaudum** [comment]
hi @devoncrouse 
this is actually a bug that you cannot insert the object array into the desc_urls table. due to refactoring this has already been fixed in master, but we'll provide a fix for 0.45.x and 0.46.x

@globalspin 
it is not possible to cast an object array (`urls`) to string, also not an object array to string array.
crash however displays objects and arrays as JSON strings

**Comment by seut** [comment]
@devoncrouse we've just released 0.45.9 which is fixing this issue (docker image still pending). A 0.46.x release will follow soon, we'll update this issue as soon as it is done.

**Comment by seut** [comment]
@devoncrouse we've also released 0.46.5 (testing) fixing this, and a 0.47 (testing) which never had this issue.
